tests/Integration/Behaviour/Features/Context/Domain/TaxFeatureContext.php

```diff
@@ -66,9 +66,12 @@ public function __construct()
      */
     public function createTax(string $taxReference, TableNode $table): void
     {
-        $data = $table->getRowsHash();
+        // we don't want to duplicate taxes if they already exist
+        if (SharedStorage::getStorage()->exists($taxReference)) {
+            return;
+        }
```

Comment:
```suggestion
```

I disagree there are many use cases in our behat tests where we use the same reference several times, it's probably not the best idea but regardless if a reference is used multiple times to create an entity the it should be created regardless and replaced